### DISCUSSION TITLE
Implement Reflection in COSMOS

### DISCUSSION THREAD

**Discussion by SirBrahms** [CosmosOS/Cosmos#2185] [Ideas]
I hereby submit the idea of implementing Reflection in Cosmos, because I'm too stupid to do it myself.

**Comment by zarlo** [comment]
it would be safe to say that is years away

**Reply by SirBrahms** [reply]
I figured that; I just thought I'd throw in the idea

**Comment by quajak** [comment]
Implementing reflection would be a major effort but in the update to .net 5 some work has been done on it to support the parts needed by runtime. It seems to me, that we can make continuous improvements to reflection support if we have a list of usecases we want to support and have someone who is willing to work on it. At some point, we might need to consider if we want to rework how we load in the data and read it at runtime from some file rather than including them directly in the binary.

**Reply by MishaTy** [reply]
That's what unity does with IL2CPP, storing metadata in a file

**Reply by retrievaaaa** [reply]
Reflection would be awesome, it would probably make using something like [ILRuntime ](https://github.com/Ourpalm/ILRuntime/) possible (if not, it would probably need some tiny modifications), which would therefore allow to run C# code from a VFS file
